apps/sms/js/threads.js

```diff
@@ -86,12 +88,19 @@
 
   var Threads = exports.Threads = {
     registerMessage: function(message) {
-      var thread = Thread.create(message);
       var threadId = message.threadId;
+
       if (!this.has(threadId)) {
-        this.set(threadId, thread);
+        this.set(threadId, Thread.create(message));
+      }
+      this.get(threadId).messages.set(message.id, message);
+    },
+    unregisterMessage: function(message) {
```

Comment:
But how can we delete the message by message ID only? we will still need threadId at least to fetch the thread contains the message(unless we assume it's active thread)